### DISCUSSION TITLE
apache-airflow-providers-oracle issue test connection

### DISCUSSION THREAD

**Discussion by sonnn1505** [apache/airflow#33278] [General]
I have got an issue about connection provider in airflow to oracle DB like below screen-sort. Please kindly give me the solution to invest this err.

![image](https://github.com/apache/airflow/assets/32322668/4dbdc037-0652-4ff4-91f3-05bc1c93831b)

**Comment by potiuk** [comment]
Not a solution but advice how you can look further:

look at the webserver log for details if the detail is not enough. That's the best advice I can give. And read Oracle's documentation or try to google "Error 22 Oracle". Apparently this is the error returned by the driver and one of your arguments is invalid, but if you do not have details enouhg in the error returned, you need to dig deeper.

Maybe also look at the source code of the Oracle Hook and that might give you some hints 

Good luck!

**Reply by sonnn1505** [reply]
Thank for your advice.
I have downgrade from **apache-airflow-providers-oracle==3.x.x** to **apache-airflow-providers-oracle==2.2.2,**
worked for me!

**Reply by gulstein86** [reply]
I'm using Oracle Oracle Database 19c Enterprise. This downgrade work for me. Thanks @sonnn1505

**Comment by ertancelik** [comment]
I have a question about Ariflow oracle connection. I tried a lot but couldn't find a solution, I need your help. I am using Airflow 2.5.1 version on Ubuntu machine, I have installed oracle provider etc. everything is available in the db network, but when connecting, I get this error:__init__() missing 1 required positional argument: 'backend'
<img width="2372" alt="Untitled" src="https://github.com/apache/airflow/assets/81710528/849114ea-12b8-483d-bfa4-69ded96cdd49">

**Reply by sonnn1505** [reply]
which oracle provider version, have you using? have  you ever try to downgrade oracle  provider?

**Reply by ertancelik** [reply]
Hi,
The package versions I use are as follows:

Name: oracledb Version: 1.3.1
Name: apache-airflow Version: 2.5.1
Name: apache-airflow-providers-oracle Version: 3.7.0
Name: apache-airflow-providers-common-sql Version: 1.3.3

<img width="521" alt="oracle_conn" src="https://github.com/apache/airflow/assets/81710528/baee40ff-62fb-43bc-82a8-89908c82f0b3">

**Reply by sonnn1505** [reply]
can you try to use : apache-airflow-providers-oracle Version: **2.2.2**

**Reply by ertancelik** [reply]
Version: 2.2.2 solved my problem, I wish I had tried it :) thanks a lot
